src/tests/test_store.py

```diff
@@ -281,3 +281,39 @@ def test_branch(is_dialect_support_backtick, monkeypatch):
             identifier
         )
     )
+
+
+@pytest.mark.parametrize(
+    "dependencies, expected_output",
+    [
+        (None, None),
+        (["first_a"], {"first_a": "SELECT * FROM a WHERE x > 10"}),
+        (
+            ["second_a"],
+            {
+                "first_a": "SELECT * FROM a WHERE x > 10",
+                "second_a": "SELECT * FROM first_a WHERE x > 20",
+            },
+        ),
+        (
+            ["third_a"],
+            {
+                "first_a": "SELECT * FROM a WHERE x > 10",
+                "second_a": "SELECT * FROM first_a WHERE x > 20",
+                "third_a": "SELECT * FROM second_a WHERE x > 30",
+            },
+        ),
+        (["fourth_a"], UsageError),
+    ],
+)
+def test_get_dependency_queries(dependencies, expected_output):
+    store = SQLStore()
+    store.store("first_a", "SELECT * FROM a WHERE x > 10")
+    store.store("second_a", "SELECT * FROM first_a WHERE x > 20", with_=["first_a"])
+    store.store("third_a", "SELECT * FROM second_a WHERE x > 30", with_=["second_a"])
+
+    if expected_output == UsageError:
```

Comment:
do not use if statements in unit tests; better to break them down into two tests: one for the scenario where it works and another one when it raises an error: https://ploomber-contributing.readthedocs.io/en/latest/contributing/submitting-pr.html#testing-guidelines